test/ng/compileSpec.js

```diff
@@ -3872,6 +3872,51 @@ describe('$compile', function() {
           ]);
         });
       });
+
+      it('should work if $doCheck is provided in the constructor', function() {
```

Comment:
The test above assigns the methods on the prototype. This test defines them on the instance from inside the constructor.
(Note that while both work in ng1, in ng2 only the prototype will work.)